docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md

```diff
@@ -0,0 +1,312 @@
+---
+id: ec2-io-stress
+title: EC2 IO Stress
+---
+
+## Introduction
+
+- EC2 IO Stress contains chaos to disrupt the state of infra resources. The experiment can induce a stress chaos on AWS EC2 Instance using Amazon SSM Run Command, this is carried out by using SSM Docs which is in-built in the experiment for the give chaos scenario.
+- It causes IO Stress chaos on EC2 Instance using an SSM doc for a certain chaos duration.
+
+:::tip Fault execution flow chart
+![EC2 IO Stress](./static/images/ec2-io-stress.png)
+:::
+
+## Uses
+
+### Uses of the experiment
+
+:::info
+
+- Filesystem read and write is another very common and frequent scenario we find with processes/applications that can result in the impact on its delivery. These problems are generally referred to as "Noisy Neighbour" problems.
+- Injecting a rogue process into a target ec2 instance, we starve the main processes/applications (typically pid 1) of the resources allocated to it (where limits are defined) causing slowness in application traffic or in other cases unrestrained use can cause instance to exhaust resources leading to degradation in performance of processes/applications present on the instance. So this category of chaos experiment helps to build the immunity on the application undergoing any such stress scenario.
+
+:::
+
+## Prerequisites
+
+### Verify the prerequisites
+
+- Ensure that Kubernetes Version >= 1.17
+
+**AWS EC2 Access Requirement:**
+
+- Ensure that SSM agent is installed and running in the target EC2 instance.
+- Ensure to create a Kubernetes secret having the AWS Access Key ID and Secret Access Key credentials in the `CHAOS_NAMESPACE`. A sample secret file looks like:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cloud-secret
+type: Opaque
+stringData:
+  cloud_config.yml: |-
+    # Add the cloud AWS credentials respectively
+    [default]
+    aws_access_key_id = XXXXXXXXXXXXXXXXXXX
+    aws_secret_access_key = XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+```
+
+- If you change the secret name then please also update the `experiment.yml` ENV values for deriving the respective data from the secret. Also account for the path at which this secret is mounted as a file in the manifest ENV `AWS_SHARED_CREDENTIALS_FILE`.
```

Comment:
```suggestion
## Prerequisites
:::info
- Ensure that Kubernetes Version >= 1.17
- Ensure that SSM agent is installed and running in the target EC2 instance.
- Ensure to create a Kubernetes secret having the AWS Access Key ID and Secret Access Key credentials in the `CHAOS_NAMESPACE`. A sample secret file looks like:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: cloud-secret
type: Opaque
stringData:
  cloud_config.yml: |-
    # Add the cloud AWS credentials respectively
    [default]
    aws_access_key_id = XXXXXXXXXXXXXXXXXXX
    aws_secret_access_key = XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```
- If you change the secret name then please also update the `experiment.yml` ENV values for deriving the respective data from the secret. Also account for the path at which this secret is mounted as a file in the manifest ENV `AWS_SHARED_CREDENTIALS_FILE`.
:::
```

Reply:
Can we correct the format?